### ISSUE TITLE
Making the Arrow actually fast.

### ISSUE THREAD

**Issue by Hyugat** [endless-sky/endless-sky#4567]
Currently, the Arrow serves hardly any purpose within the game. I propose that we reduce the Arrow's drag (and possibly increase engine capacity as well) in order to give it some sort of purpose, at least a fun factor for the player using it.   Currently, the arrow's description states the following: "The Arrow is a light transport, designed for bringing a small number of very rich passengers - Syndicate CEOs, for example - from one planet to another with speed and style. It makes up for its light armament by **being able to outrun nearly every other ship in the sky."**

Despite its description specifically noting that it is a speedy ship, the arrow can be outrun by the following: any interceptor with similar engines (some of which exist already in the game as variants), a Clipper with atomic engines, a falcon with atomic engines, and a **stock** Sparrow (a very common ship for pirates to use). I know it says "nearly every other ship", but this is a little too slow for my liking, and many other players would likely agree.

Does anyone in the es universe really think that important Syndicate CEOs should be transported in a ship that is outrun by common pirates? Should a ship engineered specifically for speed be slower than a common interceptor by default? Should the Syndicate's most important leaders really be so unsafe?

I propose that we decrease the value for the Arrow's drag (along with all the marauder variants) down to 2.0, from 2.7, making it faster than interceptors with the same engines, but still slower than certain ships: Some of the marauder engines variants, a Bounder with atomic engines, and the Flivver. This way its description would still be perfectly accurate, and it would be at least a little more useful. This change buffs the arrow's stock max speed from around 606 to 819, making it quite a speedy ship, but not unreasonably so. The Marauder engines Arrow's speed goes from around 1057, to a whopping 1428. (The min-maxed ship is still slower than a min-maxed Marauder Fury, the ship with the highest potential for speed in the game.) This makes these ships more fun to use, but doesn't in my opinion detract at all from the game in any way (it's not like this ship is OP, and it fits within the lore).

 If we want to further increase the Arrow's fun factor, we can buff the engine space from 60 to 70, or even 80 (buffing the Marauder Arrows as well, making them even more fun to zoom around in), but this change comes with a few issues (creating the new fastest ship in the game and such, making the engines variant be able to reach even more ridiculous speeds, etc.).

TLDR: Reduce the Arrow's drag (and maybe buff engine space) in order to to make it have at least some sort of niche or fun factor.

**Comment by Zitchas** [comment]
I like the idea, but jumping from a speed of 606 to 819 seems a bit extreme. Just looking at the current stock configurations of human ships, the arrow is currently faster than all except:
- Sparrow,
- Flivver,
- Combat Drone,
- Surveillance Drone

So, sounds to me like its description of "faster than almost every ship in the sky" is quite accurate. Personally, it doesn't seem reasonable that a civilian transport ship is faster (stock) than a military ship that was expressly designed to run down fleeing ships (this is, basically, what interceptors are designed to do). 

My suggestion is change the drag resistance to 2.6, which will result in a stock speed of 630, which will allow it to outrun the combat drone. Even if it is still technically slower than a sparrow, keep in mind that it only needs to make it to a planet. And 630 vs the stock sparrow's 640 is close enough that they should be able to make it to safety unless they're caught a long ways off from their destination.

On the other hand, I fully support bumping up its engine capacity to 75. Giving players more flexibility in what they can install is something I fully support, and such a change would allow pilots to ditch the missile launcher for either bigger engines or an afterburner, which will allow it to easily outrun a stock sparrow or surveillance drone, and quite possibly even a flivver (an afterburner almost doubles the thrust of the stock engine, so that should put it into the 1100-1200 range, briefly)

Note: I'm not comparing to marauders and variants because those are supposed to be unusual, heavily modified ships. That's like comparing stock sports cars with the frankenstein things that good hotrodding shops can accomplish with a base car and huge piles of cash and material.

**Comment by Zitchas** [comment]
Alright, take that PR for a spin and see how it feels. Should be a nice improvement, although admittedly nowhere near as fast as you were asking for.

**Comment by Amazinite** [comment]
In terms of stock outfits, the Arrow *is* among the fastest, but if we're looking at capabilities instead of stock or variant loadouts, then the Arrow is eclipsed by many ships. Looking at `engine space / drag` as a shorthand for speed (which isn't quite perfect given that larger engine space means being able to fit more efficient engines), the *Falcon* of all ships is slightly better, and 26 other human ships are better by this metric as well (excluding Marauders).
Something of note, the Arrow and Wasp share the same engine space, yet the drag of the Wasp is less than half that of the Arrow (2.7 vs 1.2). It might be worth having a look at all human transport ships given their relative uselessness outside of the early game, where as small freighters can continue to be useful outside of the early game (bulk freighting cargo for a low price), and combat ships are the ships more designed to be replaced as you discover/unlock better ships.

**Comment by Hyugat** [comment]
Thing is, no other human ship has atomics stock besides the flivver, and look how fast that is. 2.6 drag is still far too much, and its maximum speed capabilities do not match its appearance at all, or its name.

You should think of speed when you think of the Arrow, yet many non-combat focused ships easily run circles around it if they are equipped with atomics. It needs significantly less drag, more engine space, or both. The former is the easiest to implement.

**Comment by Zitchas** [comment]
I'm reluctant to make significant changes to the drag, because it is a very coarse and artificial way of affecting speed. If people in general (specifically, people such as Amazinite who generally have a better perspective of balance in core human space) think it needs to be dropped, so be it. All in all, though, I'd prefer to make changes such as engine space that allow the player to make more choices about going faster, rather than change drag which doesn't really affect player choice at all.

**Comment by Hyugat** [comment]
Again, the problem is that with other ships, the player can choose to have faster engines (atomics), with the arrow, a250 is just about the fastest you can reasonably cram into it in human space. The Arrow is essentially, already upgraded (you could cram an afterburner with the engine space change, but I've never seen a single player use those). With the drag reduction, the player can choose to create a hotrod with hai engines or afterburners and all that, or stick with the normal engine loadout, and still get their money's worth in speed. The other problem is that the drag is (at least in my opinion) disproportionately low, too much for its size, description, and appearance make it seem. It doesn't feel like an artificial boost when it's already far too slow. This ship feels like it doesn't reach a fast enough top speed for an Arrow, just look at the thing, it's clearly supposed to be able to go very fast. The turning rate is absurdly high, yet it barely is faster than an interceptor. Addressing your point on what other people think, perhaps I can get Amazinite to run a poll on the Discord and see what the majority of the playerbase thinks, and also just ask the more experienced contributors, the ones who would have a good knack for balance, as you said.

**Comment by Nescio0** [comment]
As pointed out earlier, what matters for maximum velocity is the engine capacity to drag ratio:
```
Surveillance Drone     :  28 /  0.53 = 52.8
Wasp                   :  60 /  1.2  = 50.0
Finch                  :  40 /  0.83 = 48.2
Dagger                 :  30 /  0.63 = 47.6
Combat Drone           :  28 /  0.59 = 47.5
Boxwing                :  28 /  0.59 = 47.5
Flivver                :  45 /  1.0  = 45.0
Sparrow                :  40 /  0.9  = 44.4
Lance                  :  30 /  0.7  = 42.9
Fury                   :  60 /  1.6  = 37.5
Shuttle                :  60 /  1.7  = 35.3
Hawk                   :  70 /  2.1  = 33.3
Headhunter             :  80 /  2.6  = 30.8
Bounder                : 110 /  3.7  = 29.7
Berserker              :  65 /  2.2  = 29.5
Scout                  : 100 /  3.4  = 29.4
Gunboat                :  90 /  3.1  = 29.0
Corvette               : 100 /  3.5  = 28.6
Heavy Shuttle          :  60 /  2.1  = 28.6
Barb                   :  22 /  0.8  = 27.5
Raven                  : 100 /  3.7  = 27.0
Quicksilver            :  70 /  2.7  = 25.9
Splinter               : 100 /  4.0  = 25.0
Kestrel (More Engines) : 240 / 10.0  = 24.0
Aerie                  :  95 /  4.1  = 23.2
Falcon                 : 150 /  6.7  = 22.4
Firebird               : 100 /  4.5  = 22.2
Arrow                  :  60 /  2.7  = 22.2
Blackbird              : 110 /  5.0  = 22.0
Clipper                :  80 /  3.7  = 21.6
Osprey                 : 130 /  6.1  = 21.3
Manta                  : 100 /  4.7  = 21.3
Mule                   : 110 /  5.7  = 19.3
Frigate                : 100 /  5.2  = 19.2
Kestrel (More Shields) : 210 / 11.0  = 19.1
Kestrel (More Weapons) : 210 / 11.0  = 19.1
Star Barge             :  40 /  2.1  = 19.0
Dreadnought            : 190 / 10.1  = 18.8
Star Queen             : 100 /  5.5  = 18.2
Cruiser                : 170 / 10.1  = 16.8
Leviathan              : 120 /  7.6  = 15.8
Vanguard               : 120 /  8.0  = 15.0
Argosy                 :  80 /  5.9  = 13.6
Rainmaker              :  50 /  3.8  = 13.2
Modified Argosy        :  80 /  6.2  = 12.9
Freighter              :  70 /  5.6  = 12.5
Carrier                : 210 / 17.7  = 11.9
Bastion                : 120 / 10.3  = 11.7
Bactrian               : 180 / 16.1  = 11.2
Protector              : 100 / 10.3  =  9.7
Behemoth               :  90 / 11.7  =  7.7
Hauler                 :  80 / 10.5  =  7.6
Nest                   :  80 / 10.5  =  7.6
Hauler II              :  80 / 11.5  =  7.0
Roost                  :  80 / 11.5  =  7.0
Hauler III             :  80 / 12.5  =  6.4
Skein                  :  80 / 12.5  =  6.4
Bulk Freighter         :  85 / 15.8  =  5.4
Class C Freighter      :  85 / 16.4  =  5.2
```
Smaller ships ought to have higher ratios, because larger engines tend to be more efficient.

You might want to look at ships of other species too, e.g. Remnant:
```
Starling  :  94 / 3.8 = 24.7
Pelican   : 144 / 7.2 = 20.0
Gascraft  :  34 / 1.7 = 20.0
Gull      : 101 / 6.0 = 16.8
Albatross : 142 / 8.6 = 16.5
```

**Comment by Zitchas** [comment]
No, smaller ships should not have higher ratios. Smaller ships should have lower drag, period. The ratio itself is dependent on a lot of other factors, especially "how much of a priority is engine capacity to the ship designers."

Having smaller ships have higher ratios as a rule is only the case if the designers of ships always had the same priorities of balancing engines versus other ship systems. Obviously, they don't. 

And you'll note that some of the key ships that the Arrow is being compared against (Shuttle and Wasp, for instance) are both less than half the size of the Arrow. Given that ES treats space as if aerodynamics is a thing, a smaller object with the same amount of thrust is going to go faster than a larger object. And the Arrow is roughly double the mass of the shuttle, going by hull size alone.

And this is strictly a human-space comparison here. Once players get into alien ships, then I fully assume that they are going to find better stuff to fly. 

I think the crux of the issue is in baseline terms of capacities. Namely, I don't think that comparing "What is the absolute fastest speed possible this ship could conceivably reach" is a good way of comparing ship speeds. That ratio doesn't take into account that on some of those ships it is powering said ship with maxed out engines is really hard. 

For example, take the shuttle. One *can* fit a Biroo atomic thruster on there which predictably takes its top speed up to 1400+, although that leaves it with the tiniest Basrem steering, and relies on six photovoltaic panels to provide power. Sure, it can go like the blazes for a few seconds. But then it has to sit around for what feels like an eternity waiting for batteries to recharge, meanwhile it is effectively reduced to a top speed of "coasting with ocassional spurts of thrust." Compared to the Arrow (with slight space enhancements) that can go roughly the same speed for as long as fuel lasts - and then continue to go 655 basically forever (it has more energy generation than those engines need, as far as I can see in my quick test). So, definitely a mobility advantage to the Arrow, even if the shuttle can theoretically keep up for a few seconds. 

On a housekeeping note, how about we keep all the discussion about how much faster the arrow should be (and the means of acheiving that) in this thread, and restrict the PR thread to comments just about the PR proper (the code itself.)

**Comment by Nescio0** [comment]
> No matter how well built a ship is, no matter how "optimized for speed" it might be, the hard physics dictate that if you have the same thrust on a ship that has half the mass, it is going to go faster, roughly twice as fast.. This is made up for in the fact that the Arrow has (now) 75 tons of engine space and 180 tons of outfit space versus the shuttle's 120 and 60. (edit: More accurately, all ships should have the same near-speed-of-light top speed, but the lighter ship should get there roughly twice as fast)

That's not how it works in this game. The mechanics are simple:
* maximum velocity = thrust / drag (i.e. independent of mass)
* acceleration = thrust / mass (i.e. independent of drag)
* rotation = turn / mass (i.e. independent of drag)

> If you want to convince me that the arrow needs lower drag, put together actual facts (not theoretical drag/engine space ratios) about how fast other ships can go.

The Arrow, Fury, Heavy Shuttle, Shuttle, and Wasp all have exactly the same engine capacity, which means they can all install exactly the same engines. If you install the Arrow's A250+A255 in those other ships, you'll see how high their maximum velocities are compared to the Arrow:
```
Wasp:          2.7 / 1.2 = 2.25×, i.e. 125% faster
Fury:          2.7 / 1.6 = 1.6875×, i.e. 69% faster
Shuttle:       2.7 / 1.7 = 1.5882×, i.e. 59% faster
Heavy Shuttle: 2.7 / 2.1 = 1.2857×, i.e. 29% faster
```
> Smaller ships ought to have higher ratios, because larger engines tend to be more efficient.

Let's have a look at the thrust per mass for the human atomic thrusters:
```
A120 :  15.4 /  22 = 0.7
A250 :  27.3 /  34 = 0.8
A370 :  47.6 /  53 = 0.9
A520 :  81.9 /  82 = 1.0
A860 : 139.7 / 127 = 1.1
```
and turn per mass for their steering counterparts:
```
A125 :  392 /  16 = 24.5
A255 :  687 /  25 = 27.5
A375 : 1192 /  38 = 31.4
A520 : 2050 /  60 = 34.2
A865 : 3509 /  92 = 38.1
```
As you can see in my previous post, the Arrow, Firebird, and Falcon have approximately the same engine capacity to drag ratio:
```
Falcon                 : 150 /  6.7  = 22.4
Firebird               : 100 /  4.5  = 22.2
Arrow                  :  60 /  2.7  = 22.2
```
However, the Arrow has space for A250+A255 (59/60), the Firebird for A370+A375 (91/100), and the Falcon for A520+A525 (142/150), which result into the following maximum velocities:
```
Firebird: (47.6/27.3) / (4.5/2.7) = 1.05
Falcon: (81.9/27.3) / (6.7/2.7) = 1.21
```
As you can see, *for the same engine to drag ratio*, the Firebird is about 5% and the Falcon about 21% faster than the Arrow, i.e. smaller ships are slower. Therefore I say it again:
> Smaller ships ought to have higher ratios, because larger engines tend to be more efficient.

The Arrow is supposed to be a fast ship; no disagreement on that, I hope? Right now it is actually the second slowest human transport; only the Star Queen is worse.

**Comment by Nescio0** [comment]
> Then post configurations with Top speed, Top boosted speed, and how long it can sustain both. (Limited to human tech only).

Here you go:
```
ship Wasp
	outfits
		Hyperdrive
		"A255 Atomic Steering"
		"A250 Atomic Thruster"
		"D14-RN Shield Generator"
		"Dwarf Core"
		"Water Coolant System"
```
![Screenshot from 2019-10-04 17-35-11](https://user-images.githubusercontent.com/21634324/66220632-a7d5c680-e6cd-11e9-914a-ef4786ba29b9.png)

The Arrow has a maximum speed of 606.7 *with the same engines*. This Wasp is 1365/606.7=2.25× as fast, as predicted above.

**Comment by Zitchas** [comment]
Uh huh. You can argue "But its built for speed!" all you want - If you've got two ships using the same engine, and one is nothing but a tin can big enough for two people, and the other one is a full luxury suite for five - I'm going to side with the first one over the second one for speed every single time. They're the same tech level using the same engine. This is basically along the same lines as arguing that the Bombardier Leerjet 35 should be able to outrun an F-16 because it cost a lot and was built for VIPs. 

Also, you seem overly focused on the first part of the Arrow's description, and forgetting the last part: " by being able to outrun *nearly* every other ship in the sky." 

1) If you aren't used to marketing speak, they are *never* going to take into account aftermarket modifications - They are advertising to VIPs, and will always be talking about stock configurations. You can't expect their marketing department to take into account that someone out there may sacrifice everything else on their ship to strap a slightly bigger engine on to go just that much faster than stock.

2) It also implies that there *are* ships that can go faster than it, stock. More than one, even. And even ones that can do so *and* have a weapon to threaten it with. Descriptions of capabilities are based around it (and every other ship's) stock equipment unless specifically stating otherwise.

And all that aside: Like I said, I'm happy to entertain the idea of giving it a tad more outfit space or maybe even bumping up the engine capacity a bit more. Those give the player more choices. I disagree that the drag should be reduced any farther. That doesn't give any extra player choice, just a static upgrade.

**Comment by Hyugat** [comment]
Regarding how ES treats aerodynamics, look at the Arrow's **appearance**. It's basically a long stick with engines on the back, it has a very sleek, very aerodynamic appearance, it does not remind me of a business Jet, that would go to the Blackbird or Bounder (The Bounder has a higher speed potential than the Arrow). Secondly, looking at the mass of the ship, it has less mass by default than both the Hawk and Berserker, yet those ships have much less drag, despite the Arrow looking like it would be more Aerodynamic, and weighing less than them. I think a change from 2.7 to 2.6 simply isn't enough. It actually does weigh less and appears to be more aerodynamic than some of the interceptors. 

That said, I can still see that giving it more outfit space or other buffs (bunks or shields perhaps) would be a welcome change that adds some form of usefulness to this ship (Even though for the most part I prefer drag reduction). I actually think better shields would be a good change, as it would serve to better protect the important people it's transporting. Either way, in its current state, I don't believe the PR is enough for the Arrow.

**Comment by Zitchas** [comment]
Just to note: I did reduce the drag further down to 2.5, so it does surpass the sparrow in straight stock to stock speed comparison.

**Comment by Zitchas** [comment]
OK, on Amazinite's suggestion, the Arrow's drag has been dropped to 2, so its top speed is now just over 800. It's fuel has also been increased to 600. (this is in addition to the extra engine space change).

**Comment by Hyugat** [comment]
All relevant changes have been merged, I see no reason to keep the issue open.
